api_tests/institutions/views/test_institution_summary_metrics.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+import datetime
+import random
+
+from api.base.settings.defaults import API_BASE
+from osf_tests.factories import (
+    AuthUserFactory,
+    InstitutionFactory
+)
+from osf.metrics import InstitutionProjectCounts
+
+
+@pytest.mark.es
+@pytest.mark.django_db
+class TestInstitutionSummaryMetrics:
+
+    @pytest.fixture()
+    def institution(self):
+        return InstitutionFactory()
+
+    @pytest.fixture()
+    def user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def admin(self, institution):
+        user = AuthUserFactory()
+        group = institution.get_group('institutional_admins')
+        group.user_set.add(user)
+        group.save()
+        return user
+
+    @pytest.fixture()
+    def url(self, institution):
+        return f'/{API_BASE}institutions/{institution._id}/metrics/'
+
+    def test_get(self, app, url, institution, user, admin):
+        # Tests unauthenticated user
+        resp = app.get(url, expect_errors=True)
+        assert resp.status_code == 401
+
+        # Tests unauthorized user
+        resp = app.get(url, auth=user.auth, expect_errors=True)
+        assert resp.status_code == 403
+
+        # Tests authorized user with institution without any metrics
+        resp = app.get(url, auth=admin.auth, expect_errors=True)
+        assert resp.status_code == 404
+
+        # Record latest institutional metrics to ES
+        public_project_count_latest = int(50 * random.random())
```

Comment:
Because if testmon caching there's not really much of a point to randomly generating these, it will just be cached as the same result for subsequent runs. It's also just not really conventionally what we do for tests, we try to use fixed values unless it is data like usernames that have to be unique. 

Also you'd probably want to use `random.randint` instead of this.

Reply:
Changed to use specified values